scheduler/pkg/store/experiment/store.go

```diff
@@ -461,3 +470,24 @@ func (es *ExperimentStore) GetExperiments() ([]*Experiment, error) {
 	}
 	return foundExperiments, nil
 }
+
+func (es *ExperimentStore) cleanupDeletedExperiments() {
+	es.mu.Lock()
+	defer es.mu.Unlock()
+	es.logger.Info("cleaning up deleted experiments")
+	for _, experiment := range es.experiments {
+		if experiment.Deleted {
+			if experiment.DeletedAt.IsZero() {
+				experiment.DeletedAt = time.Now()
+				if es.db != nil {
+					err := es.db.save(experiment)
+					if err != nil {
+						es.logger.Warnf("could not update DB TTL for experiment: %s", experiment.Name)
+					}
+				}
+			} else if experiment.DeletedAt.Add(utils.DeletedResourceTTL).Before(time.Now()) {
+				es.experiments[experiment.Name] = nil
```

Comment:
should the style be `delete(es.experiments, experiment.Name)`?

Reply:
yep... updated!